src/ctx.rs

```diff
@@ -121,12 +123,6 @@ impl Default for StrCtx {
     }
 }
 
-impl From<u8> for StrCtx {
```

Comment:
Length takes a usize, delim a u8. But I see your point will be ambiguous to a reader perhaps or will bite you if not careful. On that note I thought we would allow str reading via length by simply passing usize? I think this way is really cool and reasonable